subsys/shell/shell_ops.c

```diff
@@ -161,16 +161,16 @@ void z_shell_op_cursor_word_move(const struct shell *sh, int16_t val)
 
 void z_shell_op_word_remove(const struct shell *sh)
 {
-	char *str = &sh->ctx->cmd_buff[sh->ctx->cmd_buff_pos - 1];
-	char *str_start = &sh->ctx->cmd_buff[0];
-	uint16_t chars_to_delete;
-
 	/* Line must not be empty and cursor must not be at 0 to continue. */
 	if ((sh->ctx->cmd_buff_len == 0) ||
 	    (sh->ctx->cmd_buff_pos == 0)) {
 		return;
 	}
 
+	char *str = &sh->ctx->cmd_buff[sh->ctx->cmd_buff_pos - 1];
+	char *str_start = &sh->ctx->cmd_buff[0];
+	uint16_t chars_to_delete;
```

Comment:
Is it ok to declare these mid way through the function outside of any control block? Can't recall if the coding style mention it but I don't see it done anywhere else in this file.